First beta release

jpcoenen opened the pull request.

Marton6 committed:
Add xgo submodule

Marton6 committed:
Generate python module and add Makefile

Additionally, the `time` type in secrethub-xgo must be renamed as
it collides with a python type.

Marton6 committed:
Add ExportEnv method

Marton6 committed:
Add getter for time fields that converts to datetime

Marton6 committed:
Make secret and secret version fields immutable

Marton6 committed:
Remove unneeded typemap

Marton6 committed:
Add getter for uuid fields that converts to python uuid

Marton6 committed:
Move PyDateTime_IMPORT macro call to init function of module

The PyDateTime_IMPORT macro initializes the PyDateTime package
so that it can be used to convert a timestamp to a python
datetime object. This only has to be called once so it should
be placed in the init function of the python module.

Marton6 committed:
Convert result of resolveEnv to a dict

Marton6 committed:
Rename fields and methods to snake case

Renamed the symbols to follow the PEP8 style guide.
More specifically all methods and members are now camel case,
class names and constructors are in upper camel case.

Marton6 committed:
Rename package to secrethub to follow python naming

Marton6 committed:
Run make clean

Marton6 committed:
Add test github action

Marton6 committed:
Trigger github action

Marton6 committed:
Trigger github action

Marton6 committed:
Trigger github action

Marton6 committed:
Trigger github action

Marton6 committed:
Trigger github action

Marton6 committed:
Trigger github action

Marton6 committed:
Trigger github action

Marton6 committed:
Update secrethub-xgo (renamed time type)

Marton6 committed:
Add windows building steps to makefile and action

Marton6 committed:
Fix typo

Marton6 committed:
Fix windows compilation command

Marton6 committed:
Use make lib-win

Marton6 committed:
Do not use mingw

Marton6 committed:
Fix swig.exe

Marton6 committed:
Do not generate swig code

Marton6 committed:
Try mingw for linking

Marton6 committed:
Try cl.exe with preprocessor definition

Marton6 committed:
Try mingw

Marton6 committed:
Fix typo

Marton6 committed:
Try mingw for compiling and cl.exe for linking

Marton6 committed:
Try linking with link.exe

Marton6 committed:
Fix typo

Marton6 committed:
Fix command for renaming file

Marton6 committed:
Fix command for renaming file

Marton6 committed:
Use mingw for linking

Marton6 committed:
Fix typo

Marton6 committed:
Add correct linking to python

Marton6 committed:
Use pkg-config

Marton6 committed:
Try to rebuild python library

Marton6 committed:
Fix dlltool command

Marton6 committed:
Try recompiled python lib

Marton6 committed:
Try directly compiling python lib

Marton6 committed:
Build Client.dll in ubuntu environment

Marton6 committed:
Add dependency between workflows

Marton6 committed:
Do not use Makefile

Marton6 committed:
Add libpath that points to python

Marton6 committed:
Set LIBPATH environment variable

Marton6 committed:
Fix setting LIBPATH env var

Marton6 committed:
Try passing libpath

Marton6 committed:
Rename github workflow, stages and artifacts

Marton6 committed:
Add jobs for building win 64 client

Marton6 committed:
Remove debug statements

Marton6 committed:
Rename dlls to .pyd files

Marton6 committed:
Make upload artifact actions only upload the final files

Marton6 committed:
Add Client32.h and Client64.h

These files should be generated by cgo, but cgo generates files
that are not compatible with cl.exe. For this reason I took the
files generated by cgo, edited them to also work with the win
c compiler and added them as Client32.h and Client64.h. These
are renamed and used in the pipeline.

Marton6 committed:
Try using only python3.lib and add test job

Marton6 committed:
Fix libpath flag and use python3.9

Marton6 committed:
Upload all files of the package

Marton6 committed:
Add commands for packaging win64 client in a wheel

Marton6 committed:
Remove test-win64-client job

Marton6 committed:
Add dll and pyd to package

Marton6 committed:
Fix import mistake in setup.py

Marton6 committed:
Add __all__ to __init__.py

Marton6 committed:
Bump version

Marton6 committed:
Build 32bit windows wheel

Marton6 committed:
Fix import order

Marton6 committed:
Build linux package

Marton6 committed:
Fix github action yml

Marton6 committed:
Fix github action job dependencies

Marton6 committed:
Fix include path in gcc command

Marton6 committed:
Run auditwheel in docker container

Marton6 committed:
Fix docker run commands

Marton6 committed:
Make packag not pure and fix typo

Marton6 committed:
Implement workaround for including shared libs in wheel and make it non-pure

setup.py now includes _secrethub.so/pyd and Client.dll/so and makes the
wheel platform dependent such that it can be fixed with auditwheel for linux.

Marton6 committed:
Try manylinux2010 container

Marton6 committed:
Try python3.6.8 for linux

Marton6 committed:
Fix typo

Marton6 committed:
Revert back to using python3.9 and manylinux1

Marton6 committed:
Use manylinux2014 platform tag

Marton6 committed:
Add parent dir of Client.so to LD_LIBRARY_PATH

Marton6 committed:
Append to LD_LIBRARY_PATH instead of overwriting it

Marton6 committed:
Add /usr/local/lib to LD_LIBRARY_PATH and escape $

Marton6 committed:
Upload correct directory with fixed wheels

Marton6 committed:
Add closing quote

Marton6 committed:
Bump version

Marton6 committed:
Build secrethub.i as a python extension

Marton6 committed:
Fix linux package job

Marton6 committed:
Fix linux package job

Marton6 committed:
Build wheel for win32

Marton6 committed:
Install swig in github action

Marton6 committed:
Add --plat-name flag when building 32bit windows wheel

Marton6 committed:
Use 32bit python when building 32 bit windows wheel

Marton6 committed:
Fix typo

Marton6 committed:
Build win-amd64 wheel

Marton6 committed:
Fix typo

Marton6 committed:
Add Client.dll/.so/.dylib to package libraries

Marton6 committed:
Fix typo

Marton6 committed:
Fix another typo

Marton6 committed:
Include .dll with package_data

Marton6 committed:
Rename Client.dll to Client.pyd

Marton6 committed:
Add Client.pyd as package data

Marton6 committed:
Try to include Client.pyd as package data

Marton6 committed:
Fix typo

Marton6 committed:
Fix package_data

Marton6 committed:
Try to link .dll

Marton6 committed:
Rename libClient.dll -> Client.dll

Marton6 committed:
Revert setup.py to use no external objects

Marton6 committed:
Add wheel-repair.py script for repairing windows wheels

Marton6 committed:
Install dependencies of wheel-repair.py

Marton6 committed:
Run wheel-repair in bash to have wildcard \* resolving

Marton6 committed:
Create wheelhouse dir in github actions

Marton6 committed:
Bump version

Marton6 committed:
Build linux package for python version 3.5-3.9

Marton6 committed:
Fix typo

Marton6 committed:
Build windows package for python version 3.5-3.9

Marton6 committed:
Use python3.9 when running wheel-repair.py

Marton6 committed:
Install wheel-repair.py's requirements with python3.9

Marton6 committed:
Build macos wheel

Marton6 committed:
Fix installation of delocate

Marton6 committed:
Install swig before building macos package

Marton6 committed:
Fix rename Client.so -> Client.dylib

Marton6 committed:
Fix typo

Marton6 committed:
Run delocate-list

Marton6 committed:
Try delocate-wheel

Marton6 committed:
Fix package strucure

Moved the files from the root of the package to the secrethub
subdirectory to conform the standard package structure and have
delocate work.

Marton6 committed:
Add ./secrethub to DYLD_LIBRARY_PATH

Marton6 committed:
Fix typo

Marton6 committed:
Rename macos artifact

Marton6 committed:
Build python package with new package structure

Marton6 committed:
Bump version

Marton6 committed:
Fix LD_LIBRARY_PATH env var

Marton6 committed:
Fix LD_LIBRARY_PATH env var

Marton6 committed:
Build windows python package with new package structure

Marton6 committed:
Fix typo

Marton6 committed:
Fix renaming Client32/64.g -> Client.h

Marton6 committed:
Fix location of Client.lib (and Cliend.def)

Marton6 committed:
Update wheel-repair.py

Marton6 committed:
Fix wheel-repair.py

Marton6 committed:
Upload intermediary artifact for debugging

Marton6 committed:
Fix DLL directory for wheel-repair.py

Marton6 committed:
Remove python version requirement

Marton6 committed:
Add python versions 3.5-3.9 to strategy matrix

Marton6 committed:
Remove artifact uploaded for debugging purposes

Marton6 committed:
Try macos version 10.14

Marton6 committed:
Build both for macos 10.14 and macos 10.15

Marton6 committed:
Add tests for linux, windows (x64 and x86) and macos 10.15 clients

Marton6 committed:
Rename macos jobs for consistency

Marton6 committed:
Add test job for macos 10.14 client

Marton6 committed:
Add (commented out) publish job that publishes to test pypi

Marton6 committed:
Fix unit test

Marton6 committed:
Add README

Marton6 committed:
Extend setup.py with more information

Marton6 committed:
Add test for write and exists

Marton6 committed:
Fix README extension in setup.py

Marton6 committed:
Fix rm in test jobs

Marton6 committed:
Set SECRETHUB_CREDENTIAL in test jobs

Marton6 committed:
Fix removing files in test job

Marton6 committed:
Fix removing files before running unit tests

Marton6 committed:
Run pipeline only on merge to master

Marton6 committed:
Add LICENSE file

jpcoenen commented on README.md:
```diff
@@ -1,2 +1,144 @@
+The package supports Linux, Windows (32 and 64 bit) and MacOS.
+It is built for CPython versions 3.5 and above.
+
+Make sure you have create a SecretHub account and set up a credential on your system before using the library. See the [Credential](#credential) section for more info. 
```
I think this typo is also in `secrethub-dotnet`.
```suggestion
Make sure you have created a SecretHub account and set up a credential on your system before using the library. See the [Credential](#credential) section for more info. 
```

jpcoenen commented on README.md:
```diff
@@ -1,2 +1,144 @@
+
+Make sure you have create a SecretHub account and set up a credential on your system before using the library. See the [Credential](#credential) section for more info. 
+
+> Note that passphrase protected credentials are not supported by the Python client.
```
Could we support this anyway by using the `SECRETHUB_CREDENTIAL_PASSPHRASE` variable? Because it's a bit weird if we do not allow people to test it locally.

Marton6 replied:
Yes, we plan on supporting that environment variable.

jpcoenen commented on README.md:
```diff
@@ -1,2 +1,144 @@
+Come chat with us on [Discord](https://discord.gg/EQcE87s) or email us at [support@secrethub.io](mailto:support@secrethub.io)
+
+## BETA
+This project is currently in beta and we'd love your feedback! Check out the [issues](https://github.com/secrethub/secrethub-dotnet/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc) and feel free suggest cool ideas, use cases, or improvements.
```
```suggestion
This project is currently in beta and we'd love your feedback! Check out the [issues](https://github.com/secrethub/secrethub-python/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc) and feel free suggest cool ideas, use cases, or improvements.
```

jpcoenen commented on test/test.py:
```diff
@@ -0,0 +1,24 @@
+class TestSecretHubClient(unittest.TestCase):
+
+    def test_read_string(self):
+        self.assertEqual(secrethub.Client().read_string('secrethub/xgo/python/test/test'), 'foo')
```
Do we need to embed these paths into the code? If it is possible to make them configurable, it would be easier for anyone to run these tests.

Marton6 replied:
Good point! I made them configurable.

jpcoenen reviewed (commented):
Looks very sweet @Marton6 !

![](https://media0.giphy.com/media/ZArWQGxulwNxty4ck4/giphy.gif)

florisvdg commented:
Let's make sure to squash, when merging 😅

Marton6 committed:
Fix typo 'create' -> 'created'

Co-authored-by: Joris Coenen <jpcoenen@users.noreply.github.com>

Marton6 committed:
Change 'dotnet' -> 'python'

Co-authored-by: Joris Coenen <jpcoenen@users.noreply.github.com>

Marton6 committed:
Deduplicate test jobs, simplify publish and rename artifacts

Marton6 committed:
Deduplicate build for macos jobs

Marton6 committed:
Deduplicate build win-dll jobs

Marton6 committed:
Deduplicate build-windows-client jobs

Marton6 committed:
Fix job names

Marton6 committed:
Test pipeline

Marton6 committed:
Merge branch 'feature/macos-package' of github.com:secrethub/secrethub-python into feature/macos-package

Marton6 committed:
Fix strategy.matrix.include

Marton6 committed:
Fix command that depends on arch

Marton6 committed:
Add publish job that publishes to testpypi

Marton6 committed:
Fix indentation

Marton6 committed:
Add python version list to matrix.include

Marton6 committed:
Parameterize secrethub dir used in tests

Marton6 committed:
Fix matrix of test and publish jobs

Marton6 committed:
Fix test job

Marton6 committed:
Fix test job

Marton6 committed:
Try fixing windows tests

Marton6 committed:
Update pip before installing wheel in test job

Marton6 committed:
Print compatible platform tags for debugging

Marton6 committed:
Remove printing compatible wheel tags for debugging

Marton6 committed:
Fix build command platform name for 64bit windows

Marton6 committed:
Use python3.9 for uploading wheels

Marton6 committed:
Bump version

Marton6 committed:
Use secrethub github action v0.2.1

Marton6 committed:
Fix publish command username

Marton6 committed:
Write simple script for adding Client.dll to windows wheel

Marton6 committed:
Remove print statements for debuggin

Marton6 committed:
Remove outdated Makefile

Marton6 committed:
Only run publish on push to master

Marton6 committed:
Change version number to 0.1.0

Marton6 committed:
Build wheel for all python versions >=3.5

Marton6 committed:
Uncomment win32 package test and publishing

Marton6 committed:
Publish to pypi

jpcoenen commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,212 @@
+        env:
+          SECRETHUB_CREDENTIAL: ${{ secrets.TEST_SECRETHUB_CREDENTIAL }}
+  publish:
+    if: github.event_name != 'push' && github.ref == 'refs/heads/master'
```
This means wer're publishing every commit on `master`. What happens if the version in `setup.py` is not updated and we merge a change to `master`? Does that mean we re-release an old version with a change?

Marton6 replied:
No. Twine currently fails if you try to upload a file that was already published to pypi.

jpcoenen replied:
The actual branch is currently called `main`, btw. So this is not going to work :grin:

Marton6 replied:
Oops :sweat_smile:

jpcoenen reviewed (commented):
Nice update to the pipeline! It now has a good balance between readability and code duplication :+1: And it is not much code for 50+ jobs.

I only have one question left about the publish job.

Marton6 committed:
Change referenced branch name 'master'->'main'

Marton6 committed:
Revert: Change referenced branch name 'master'->'main'

Marton6 committed:
Also build package for macos-11

Marton6 committed:
Get wheel-repair.py from another repo

jpcoenen commented on .github/workflows/build.yml:
```diff
@@ -104,7 +104,8 @@ jobs:
           architecture: ${{ matrix.arch }}
       - run: python -m pip install pefile machomachomangler
-      - run: python utilities/wheel-fix.py ./dist/*.whl ./secrethub/CLIENT.dll ./wheelhouse
+      - run: git clone https://github.com/secrethub/python-wheel-repair.git
```
As a reference: this is done to correctly attribute the original source of this script, which was published under a GPL license, which requires our modifications to also be published under a GPL license. To make this clear, this code has been published in a separate repo, as that there can be no confusion about the fact that `secrethub-python` can be published under the Apache 2.0 license.

jpcoenen commented:
Awesome!

![](https://media1.giphy.com/media/1n4FT4KRQkDvK0IO4X/giphy.gif)

I cannot give official approval because I created the PR. But approval to this PR, I give.

Marton6 committed:
Do not build for macos-11.0 from pipeline

Decided to not build the client for this macos version as the
GitHub Actions container for it is still in preview and takes a
lot of time to initialize (30 minutes or so). The client for
macos-11.0 will be published manually. The macos-11.0 should
be added back to the pipeline once it is no longer in preview.

Marton6 committed:
Publish on push to master
